hostpapa/nginx-fpm-alpine
# DEVOPS-203 - Add support for multi version image build

**Description:**
This PR adds the ability to build multiple versions of PHP/Nginx using the `versions.json` file as the main config file.

**Testing:**
- Clone this repository & switch over to this branch
- Run `bash apply-templates.sh` at the root of this repository
- You will see it will create the three versions specified in the `versions.json` file
  - In this particular case, since the folders already exists, you should not see any change.
  - If you like, remove the `8.0, 8.1, 8.2` folders and re-run the script. It will generate the folders and you shouldn't see any changes git wise.
- You can go inside each of the generated folders and do the following:
  - Build the image locally `docker build -t hostpapa/nginx-fpm-alpine .`
  - Run the image locally `docker run -p 8080:80 hostpapa/nginx-fpm-alpine`
  - Confirm the php version `docker exec -it {CONTAINERID} php -v`
  - Confirm the alpine version `docker exec -it {CONTAINERID} cat /etc/*release*`

**Screenshot:**
![Screenshot from 2023-06-09 13-19-50](https://github.com/hostpapa/nginx-fpm-alpine/assets/66637765/3bb58dfc-40d8-42f0-a265-8f31877e7a6e)
